tests/e2e/support/objects/app-admin-settings/spaces/actions.ts

```diff
@@ -49,39 +52,68 @@ export const changeSpaceQuota = async (args: {
     page.locator(actionConfirmButton).click()
   ])
 }
+export const clearSelection = async (args: { page: Page }): Promise<void> => {
+  const { page } = args
+  try {
+    await page.locator(clearSelectionBtn).click({ timeout: 500 })
+  } catch (ex) {}
+}
+export const disableSpaceButtonShouldBeGone = async (args: { page: Page }): Promise<void> => {
+  const { page } = args
+  expect(await page.locator(disableActionBtn).count()).toEqual(0)
+}
+export const disableSpace = async (args: {
+  page: Page
+  id: string
+  context: string
+}): Promise<void> => {
+  const { page, id, context } = args
+  const isBatchActions = context === 'batch-actions'
 
-export const disableSpace = async (args: { page: Page; id: string }): Promise<void> => {
-  const { page, id } = args
-  await page.locator(util.format(spaceIdSelector, id)).click()
+  if (!isBatchActions) {
+    await page.locator(util.format(spaceIdSelector, id)).click()
+  }
   await page.waitForSelector(disableActionBtn)
-  await page.locator(disableActionBtn).click()
+  await page.locator(`.${context}`).locator(disableActionBtn).click()
   await page.waitForSelector(modalConfirmBtn)
 
   await Promise.all([
     page.waitForResponse(
       (resp) =>
-        resp.url().endsWith(encodeURIComponent(id)) &&
+        (isBatchActions || resp.url().endsWith(encodeURIComponent(id))) &&
```

Comment:
Why is `resp.url().endsWith(encodeURIComponent(id)))` not needed in case of batch actions?

Reply:
Question is still valid I guess?